app/views/ThreadMessagesView/index.tsx

```diff
@@ -1,380 +1,86 @@
-				rid: this.rid,
-				count: API_FETCH_COUNT,
-				offset,
-				text: searchText
-			});
-			if (result.success) {
-				this.updateThreads({ update: result.threads, lastThreadSync });
-				this.setState({
-					loading: false,
-					end: result.count < API_FETCH_COUNT,
-					offset: offset + API_FETCH_COUNT
-				});
-			}
-		} catch (e) {
-			log(e);
-			this.setState({ loading: false, end: true });
-		}
-	}, 300);
-
-	// eslint-disable-next-line react/sort-comp
-	sync = async (updatedSince: Date) => {
-		this.setState({ loading: true });
-
-		try {
-			const result = await Services.getSyncThreadsList({
-				rid: this.rid,
-				updatedSince: updatedSince.toISOString()
-			});
-			if (result.success && result.threads) {
-				const { update, remove } = result.threads;
-				this.updateThreads({ update, remove, lastThreadSync: updatedSince });
-			}
-			this.setState({
-				loading: false
-			});
-		} catch (e) {
-			log(e);
-			this.setState({ loading: false });
-		}
-	};
-
-	onSearchPress = () => {
-		this.setState({ isSearching: true }, () => this.setHeader());
-	};
-
-	onCancelSearchPress = () => {
-		this.setState({ isSearching: false, searchText: '' }, () => {
-			const { subscription } = this.state;
-			this.setHeader();
-			this.subscribeMessages(subscription);
-		});
-	};
-
-	onSearchChangeText = debounce((searchText: string) => {
-		const { subscription } = this.state;
-		this.setState({ searchText }, () => this.subscribeMessages(subscription, searchText));
-	}, 300);
-
-	onThreadPress = debounce(
+import useThreadFilter from './hooks/useThreadFilter';
+import getFilteredThreads from './utils/getFilteredThreads';
+import useThreadSearch from './hooks/useeThreadSearch';
+
+const ThreadMessagesView = ({ navigation, route }: IThreadMessagesViewProps) => {
+	const viewName = ThreadMessagesView.name;
+	const rid = route.params?.rid;
+	let messagesObservable: Observable<TThreadModel[]> | any;
+	const [currentFilter, setCurrentFilter] = useState(Filter.All);
+	const [search, setSearch] = useState<ISearchThreadMessages>({} as ISearchThreadMessages);
+
+	const { theme } = useTheme();
+	const { user, useRealName, isMasterDetail } = useAppSelector(state => ({
+		user: getUserSelector(state),
+		useRealName: state.settings.UI_Use_Real_Name as boolean,
+		isMasterDetail: state.app.isMasterDetail
+	}));
+
+	const {
+		init,
+		initSubscription,
+		messages,
+		subscription,
+		displayingThreads,
+		loadMore,
+		loading,
+		setDisplayingThreads,
+		subscribeMessages,
+		unsubscribeMessages
+	} = useThreadMessages({
+		user,
+		messagesObservable,
+		currentFilter,
+		rid,
+		search
+	});
+
+	const { initFilter, showFilters } = useThreadFilter({
+		currentFilter,
+		setCurrentFilter,
+		user,
+		messages,
+		setDisplayingThreads,
+		subscription
+	});
+
+	const { setHeader } = useThreadSearch({
+		isMasterDetail,
+		navigation,
+		search,
+		setSearch,
+		showFilters,
+		subscribeMessages
+	});
+
+	const onThreadPress = debounce(
```

Comment:
A normal debounce function won't work inside a function component.
You should be using `useDebounce` or `useDebouncedCallback`